fcrepo-webapp/src/main/java/org/fcrepo/webapp/AuthConfig.java

```diff
@@ -0,0 +1,220 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthConfig.class);
+
+    static class HeaderPrincipalEnabled extends ConditionOnPropertyBoolean {
+        HeaderPrincipalEnabled() {
+            super(AuthPropsConfig.FCREPO_AUTH_PRINCIPAL_HEADER_ENABLED, false);
+        }
+    }
+    static class RolesPrincipalEnabled extends ConditionOnPropertyBoolean {
+        RolesPrincipalEnabled() {
+            super(AuthPropsConfig.FCREPO_AUTH_PRINCIPAL_ROLES_ENABLED, false);
+        }
+    }
+    static class DelegatePrincipalEnabled extends ConditionOnPropertyBoolean {
+        DelegatePrincipalEnabled() {
+            super(AuthPropsConfig.FCREPO_AUTH_PRINCIPAL_DELEGATE_ENABLED, true);
+        }
+    }
+
+    @PostConstruct
+    public void postConstruct() {
+        LOGGER.info("Auth enabled");
+    }
+
+    /**
+     * Optional PrincipalProvider filter that will inspect the request header, "some-header", for user role values
+     *
+     * @param propsConfig config properties
+     * @return header principal provider
+     */
+    @Bean
+    @Order(1)
+    @Conditional(AuthConfig.HeaderPrincipalEnabled.class)
+    public PrincipalProvider headerProvider(final AuthPropsConfig propsConfig) {
+        LOGGER.info("Auth header principal provider enabled");
+        final var provider = new HttpHeaderPrincipalProvider();
+        provider.setHeaderName(propsConfig.getAuthPrincipalHeaderName());
+        provider.setSeparator(propsConfig.getAuthPrincipalHeaderSeparator());
+        return provider;
+    }
+
+    /**
+     * Optional PrincipalProvider filter that will use container configured roles as principals
+     *
+     * @param propsConfig config properties
+     * @return roles principal provider
+     */
+    @Bean
+    @Order(2)
+    @Conditional(AuthConfig.RolesPrincipalEnabled.class)
+    public PrincipalProvider containerRolesProvider(final AuthPropsConfig propsConfig) {
+        LOGGER.info("Auth roles principal provider enabled");
+        final var provider = new ContainerRolesPrincipalProvider();
+        provider.setRoleNames(new HashSet<>(propsConfig.getAuthPrincipalRolesList()));
+        return provider;
+    }
+
+    /**
+     * delegatedPrincipleProvider filter allows a single user to be passed in the header "On-Behalf-Of",
+     *            this is to be used as the actor making the request when authenticating.
+     *            NOTE: On users with the role fedoraAdmin can delegate to another user.
+     *            NOTE: Only supported in WebAC authentication
+     *
+     * @return delegate principal provider
+     */
+    @Bean
+    @Order(3)
+    @Conditional(AuthConfig.DelegatePrincipalEnabled.class)
+    public PrincipalProvider delegatedPrincipalProvider() {
+        LOGGER.info("Auth delegate principal provider enabled");
+        return new DelegateHeaderPrincipalProvider();
+    }
+
+    /**
+     * Define the Shiro Realm implementation you want to use to connect to your back-end
```

Comment:
Not sure this comment makes sense anymore out of fcrepo-config.xml, maybe it should just say `WebAC Authorization Realm`?